sdk/digitaltwins/azure-digitaltwins-core/src/samples/java/com/azure/digitaltwins/core/ComponentSyncSamples.java

```diff
@@ -0,0 +1,158 @@
+import com.azure.digitaltwins.core.serialization.BasicDigitalTwin;
+import com.azure.digitaltwins.core.serialization.DigitalTwinMetadata;
+import com.azure.digitaltwins.core.serialization.ModelProperties;
+import com.azure.digitaltwins.core.util.UpdateOperationUtility;
+import com.azure.identity.ClientSecretCredentialBuilder;
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+
+import javax.net.ssl.HttpsURLConnection;
+import java.io.IOException;
+import java.util.*;
+
+public class ComponentSyncSamples {
+    private static DigitalTwinsClient client;
+    private static final ObjectMapper mapper = new ObjectMapper();
+
+    public static void main(String[] args) throws IOException, InterruptedException {
+
+        SamplesArguments parsedArguments = new SamplesArguments(args);
+
+        client = new DigitalTwinsClientBuilder()
+            .tokenCredential(
+                new ClientSecretCredentialBuilder()
+                    .tenantId(parsedArguments.getTenantId())
+                    .clientId(parsedArguments.getClientId())
+                    .clientSecret(parsedArguments.getClientSecret())
+                    .build()
+            )
+            .endpoint(parsedArguments.getDigitalTwinEndpoint())
+            .httpLogOptions(
+                new HttpLogOptions()
+                    .setLogLevel(parsedArguments.getHttpLogDetailLevel()))
+            .buildClient();
+
+        runComponentSample();
+    }
+
+    public static void runComponentSample() throws JsonProcessingException {
+
+        ConsoleLogger.printHeader("COMPONENT SAMPLES");
+
+        // For the purpose of this example we will create temporary models using a random model Ids.
+        // We have to make sure these model Ids are unique within the DT instance.
+
+        String componentModelId = UniqueIdHelper.getUniqueModelId(SamplesConstants.TemporaryComponentModelPrefix, client);
+        String modelId = UniqueIdHelper.getUniqueModelId(SamplesConstants.TemporaryModelPrefix, client);
+        String basicDigitalTwinId = UniqueIdHelper.getUniqueDigitalTwinId(SamplesConstants.TemporaryTwinPrefix, client);
+
+        String newComponentModelPayload = SamplesConstants.TemporaryComponentModelPayload
+            .replace(SamplesConstants.ComponentId, componentModelId);
+
+        String newModelPayload = SamplesConstants.TemporaryModelWithComponentPayload
+            .replace(SamplesConstants.ModelId, modelId)
+            .replace(SamplesConstants.ComponentId, componentModelId);
+
+        List<String> modelsList = new ArrayList<>(Arrays.asList(newComponentModelPayload, newModelPayload));
+
+        ConsoleLogger.printHeader("Create Models");
+        // We now create all the models (including components)
+        client.createModels(modelsList);
+
+        ConsoleLogger.print("Created models: " + componentModelId + " and " + modelId);
```

Comment:
wouldn't we want to iterate over the created models list and use those Ids to log the success statement here?